jobs: TestBatchJobsCreation failed

jobs.TestBatchJobsCreation [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9242191?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/9242191?buildTab=artifacts#/) on release-23.1 @ [6be5755ca62b8506743b0a0adde19b64135a435f](https://github.com/cockroachdb/cockroach/commits/6be5755ca62b8506743b0a0adde19b64135a435f):


```
=== RUN   TestBatchJobsCreation
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/f335d7ae74f7e66adfde72fddf47c084/logTestBatchJobsCreation2737622174
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestBatchJobsCreation
    registry_test.go:526: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/f335d7ae74f7e66adfde72fddf47c084/logTestBatchJobsCreation2737622174
--- FAIL: TestBatchJobsCreation (79.27s)
=== RUN   TestBatchJobsCreation/extra_large_batch
    registry_test.go:519: query 'SELECT count(*) FROM system.job_info WHERE info_key = 'legacy_payload'::BYTES': expected:
        5001
        
        got:
        5002
        
    --- FAIL: TestBatchJobsCreation/extra_large_batch (62.59s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestBatchJobsCreation.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25918

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -505,14 +505,18 @@ func TestBatchJobsCreation(t *testing.T) {
 
 				// Create a batch of job specifications.
 				var records []*Record
+				var jobIDStrings []string
 				for i := 0; i < test.batchSize; i++ {
+					jobID := r.MakeJobID()
+					jobIDStrings = append(jobIDStrings, fmt.Sprintf("%d", jobID))
 					records = append(records, &Record{
-						JobID:    r.MakeJobID(),
+						JobID:    jobID,
 						Details:  jobspb.ImportDetails{},
 						Progress: jobspb.ImportProgress{},
 						Username: username.RootUserName(),
 					})
 				}
+				jobIdsClause := fmt.Sprint(strings.Join(jobIDStrings, ", "))
 				// Create jobs in a batch.
 				var jobIDs []jobspb.JobID
 				require.NoError(t, ief.Txn(ctx, func(ctx context.Context, txn isql.Txn) error {
@@ -521,13 +525,13 @@ func TestBatchJobsCreation(t *testing.T) {
 					return err
 				}))
 				require.Equal(t, len(jobIDs), test.batchSize)
-				tdb.CheckQueryResults(t, "SELECT count(*) FROM [SHOW JOBS]",
+				tdb.CheckQueryResults(t, fmt.Sprintf("SELECT count(*) FROM [SHOW JOBS] WHERE job_id IN (%s)", jobIdsClause),
 					[][]string{{fmt.Sprintf("%d", test.batchSize)}})
 
 				// Ensure that we are also writing the payload and progress to the job_info table.
-				tdb.CheckQueryResults(t, `SELECT count(*) FROM system.job_info WHERE info_key = 'legacy_payload'::BYTES`,
+				tdb.CheckQueryResults(t, fmt.Sprintf(`SELECT count(*) FROM system.job_info WHERE info_key = 'legacy_payload'::BYTES AND job_id IN (%s)`, jobIdsClause),
 					[][]string{{fmt.Sprintf("%d", test.batchSize)}})
-				tdb.CheckQueryResults(t, `SELECT count(*) FROM system.job_info WHERE info_key = 'legacy_progress'::BYTES`,
+				tdb.CheckQueryResults(t, fmt.Sprintf(`SELECT count(*) FROM system.job_info WHERE info_key = 'legacy_progress'::BYTES AND job_id IN (%s)`, jobIdsClause),
 					[][]string{{fmt.Sprintf("%d", test.batchSize)}})
 			}
 		})

